### PR TITLE
ci: enable buildpulse for client tests

### PR DESCRIPTION
**Pull request by millsp**
TODO: As a bonus I'd like to move repetitive config to a separate file

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -158,6 +158,8 @@ jobs:
       TEST_FUNCTIONAL_MONGO_URI: 'mongodb://root:prisma@localhost:27018/PRISMA_DB_NAME?authSource=admin'
       TEST_FUNCTIONAL_COCKROACH_URI: 'postgresql://prisma@localhost:26257/PRISMA_DB_NAME'
       NODE_OPTIONS: '--max-old-space-size=8096'
+      JEST_JUNIT_SUITE_NAME: '${{ github.job }}/client/functional/${{ matrix.os }}/node-${{ matrix.node }}/${{ matrix.queryEngine }}'
+      JEST_JUNIT_UNIQUE_OUTPUT_NAME: true
```
Comment (millsp) [inline .github/workflows/test.yml:162]: Ensures that multiple runs do not overwrite each other's `junit.xml` outputs.

File: packages/client/tests/functional/jest.config.js [hunk 2]
```diff
@@ -18,25 +18,28 @@ module.exports = () => {
...
   }
 
+  if (process.env['JEST_JUNIT_DISABLE'] !== 'true') {
+    configCommon.reporters.push([
+      'jest-junit',
+      {
+        addFileAttribute: 'true',
+        ancestorSeparator: ' › ',
+        classNameTemplate: (vars) => {
+          return vars.classname.replace(/ \(provider=.*?\)/g, '')
```
Comment (millsp) [inline packages/client/tests/functional/jest.config.js:36]: Remove matrix information from test name

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -1330,4 +1421,4 @@ jobs:
         if: always()
         with:
           name: ${{ github.job }}_${{ matrix.os }}_node-${{ matrix.node }}_${{ matrix.queryEngine }}_junit.xml
-          path: packages/*/junit.xml
+          path: packages/*/junit*.xml
```
Comment (janpio) [inline .github/workflows/test.yml:1424]: Why the additional `*`? Does `JEST_JUNIT_UNIQUE_OUTPUT_NAME: true` influence this?

**Review by janpio — APPROVED** [review]

File: .github/workflows/test.yml (hunk 3 above)
Comment (millsp) [reply .github/workflows/test.yml:1424]: Yes, this is to make sure we match the file that now contains an id

Comment (janpio) [reply .github/workflows/test.yml:1424]: Then this should actually work - seems Buildpulse only uses the string inside to match the test, not the filename as I thought. Nice!

**Pull request merged**